image/image.go

```diff
@@ -790,7 +790,18 @@ func DecryptHwFull(img Image, secret []byte) (Image, error) {
 	return img, nil
 }
 
-// IsEncrypted indicates whether an image's "encrypted" flag is set.
+// IsEncrypted indicates whether an image's "encrypted" flag is set or hw encryption is used.
 func (img *Image) IsEncrypted() bool {
-	return img.Header.Flags&IMAGE_F_ENCRYPTED != 0
+	enc := false
+	if img.Header.Flags&IMAGE_F_ENCRYPTED != 0 {
+		enc = true
+	} else {
+		for _, tlv := range img.ProtTlvs {
+			if tlv.Header.Type&IMAGE_TLV_SECRET_ID_LEGACY != 0 || tlv.Header.Type&IMAGE_TLV_SECRET_ID != 0 {
+				enc = true
+			}
+		}
```

Comment:
`tlv.Header.Type` is a regular integer, not a bitmap.  So you should compare it with the constant directly using `==` rather than `&`.

But I think it would be easier to just use the "find" functions that already exist.  That suggestion is up to you though.

Are you sure this doesn't break anything?  I remember there being some tricky image types that used encryption, but whose encrypted flag was unset.  I recall it being important that these images appeared unencrypted during some parts of the image production process.  It has been a long time since I've looked at any of this, so I could definitely be wrong.  If you say it works, I'll take your word for it.
 
```suggestion
			if img.FindAllUniqueTlv(IMAGE_TLV_SECRET_ID_LEGACY) != nil || img.FindAllUniqueTlv(IMAGE_TLV_SECRET_ID) != nil {
			    enc = true
			}
```

Reply:
@ccollins476ad for the hardware encrypted images, we weren't setting the encrypted flag.  For those images we used a pair of TLVs describing the hw key offset to be used and a nonce.